src/mcp/tools/dataconnect/list_dataconnect_services.ts

```diff
@@ -0,0 +1,25 @@
+import { z } from "zod";
+import { tool } from "../../tool.js";
+import { mcpError, toContent } from "../../util.js";
+import { AppPlatform, getAppConfig, listFirebaseApps } from "../../../management/apps.js";
+import * as client from "../../../dataconnect/client";
+
+export const list_dataconnect_services = tool(
+  {
+    name: "list_dataconnect_services",
+    description: "List the Firebase Data Connect Services Deploy.",
+    inputSchema: z.object({}),
+    annotations: {
+      title: "List the Firebase Data Connect Services that's available in the backend",
+      readOnlyHint: true,
+    },
+    _meta: {
+      requiresProject: true,
+    },
+  },
+  async (_, { projectId }) => {
+    if (!projectId) return mcpError("No current project detected.");
+    const services = await client.listAllServices(projectId);
+    return toContent(services, { format: "yaml" });
```

Comment:
For the error handling.

Honestly, I think it should be the caller's responsibility to check for unchecked exception.

WIll send a follow-up PR to cleanup.

Reply:
I think you're probably right. I'll do a PR to do some universal error handling in the tool setup.